### PR TITLE
Added polynomial derivatives

### PR DESCRIPTION
**Pull request by jessemapel**

### CODE REVIEW

File: src/eal.cpp [hunk 1]
```diff
@@ -186,13 +195,25 @@ namespace eal {
...
     }
 
-    const double *coeffsArray = coeffs.data();
-    return gsl_poly_eval(coeffsArray, coeffs.size(), time);
+    if (d < 0) {
+      throw invalid_argument("Invalid derivative degree, must be non-negative.");
+    }
+
+    vector<double> derivatives(d + 1);
+    gsl_poly_eval_derivs(coeffs.data(), coeffs.size(), time,
```
Comment (kberryUSGS) [inline src/eal.cpp]: :+1: gsl is great! This is such a convenient function.

File: tests/EalTest.cpp [hunk 2]
```diff
@@ -136,6 +136,17 @@ TEST(SplineInterpTest, Extrapolate) {
...
+  vector<double> coeffs = {1.0, 2.0, 3.0}; // 1 + 2x + 3x^2
+  EXPECT_EQ(2.0, eal::evaluatePolynomial(coeffs, -1, 0));
+}
+
+TEST(PolynomialTest, Derivatives) {
+  vector<double> coeffs = {1.0, 2.0, 3.0}; // 1 + 2x + 3x^2
+  EXPECT_EQ(-4.0, eal::evaluatePolynomial(coeffs, -1, 1));
+  EXPECT_EQ(6.0, eal::evaluatePolynomial(coeffs, -1, 2));
+}
+
```
Comment (kberryUSGS) [inline tests/EalTest.cpp]: Add a test for `getVelocity`, including the error message if 3 vectors aren't supplied as input?

Comment (jessemapel) [reply tests/EalTest.cpp]: Will do

**Comment by jessemapel** [comment]
@kberryUSGS This should be ready for re-review now

**Pull request merged**